### PR TITLE
Enhance plan schemas

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
This PR enhance plan schemas by adding titles to enum values.
Related to #651

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
The cosmos DB (Mongo) provides a schema with 'readRegions':
```
"readRegions": {
  "type": "array",
  "title": "Read Regions",
  "description": "Read regions to be created, your data will be synchronized across these regions."
                                      },
```

The `items` field is missing. It should looks like:
```
"allowedIPRanges": {
   "type": "array",
   "title": "OneOf IP ranges",
   "description": "Values to include in IP Filter. Can be an IP Address or CIDR range.",
   "items": {
   "type": "string",
   "description": "Must be a valid IP address or CIDR"
    }
 }
```

I'd like to add missing items, but I don't know what should be added there.

File: pkg/services/servicebus/plan_schemas.go [hunk 1]
```diff
@@ -82,7 +82,11 @@ func generateTopicBindingParamsSchema() service.InputParametersSchema {
 				Description: "Specifies whether to create a subscription in the topic." +
 					" Valid values are [\"yes\", \"no\"]. ",
 				AllowedValues: []string{"yes", "no"},
```
Comment (polskikiel) [inline pkg/services/servicebus/plan_schemas.go]: You can delete AllowedValues field and adjust the description.

File: pkg/service/schema.go [hunk 2]
```diff
@@ -112,6 +112,7 @@ type StringPropertySchema struct {
 	AllowedPattern          string                        `json:"pattern,omitempty"` // nolint: lll
 	CustomPropertyValidator CustomStringPropertyValidator `json:"-"`
 	DefaultValue            string                        `json:"default,omitempty"` // nolint: lll
+	OneOf                   []EnumValue                   `json:"oneOf,omitempty"`   //nolint: lll
```
Comment (polskikiel) [inline pkg/service/schema.go]: Please add space to the `//nolint` comment to align it to the other occurences.

**Comment by norshtein** [comment]
No, they are not the same. Allowed values of Cosmos read region can be found [here](https://github.com/Azure/open-service-broker-azure/blob/master/pkg/services/cosmosdb/cosmos-utils.go#L276-L305). It is slightly different from location values.

File: pkg/services/servicebus/plan_schemas.go (hunk 1 above)
Comment (piotrmiskiewicz) [reply pkg/services/servicebus/plan_schemas.go]: Fixed

**Comment by piotrmiskiewicz** [comment]
I've added read locations to the Cosmos DB Mongo schema

**Comment by norshtein** [comment]
I prefer to keep previous `allowedValues` for some time. Although it seems to be redundant, it can provide backward compatibility. Could you keep `allowedValues` unchanged?

**Comment by zhongyi-zhang** [comment]
I think we don't need to consider backward compatibility in this case. The parsers which leverage React or  Angular won't be impacted. On the contrary, with both `AllowedValues` and `OneOf`, I am not sure whether they are impacted. And, With the announcement in release note, if the change really breaks some parsers, the user still can wait for the fix in the parser side. Make sense?

**Comment by piotrmiskiewicz** [comment]
React jsonschema (https://github.com/mozilla-services/react-jsonschema-form) does not work with both. The 'oneOf' field works very nice

**Comment by zhongyi-zhang** [comment]
Oh, good to know. Then the old should be removed for sure. @piotrmiskiewicz thanks a lot for nice contribution! I’ll merge it as the CI also passed.

**Comment by piotrmiskiewicz** [comment]
When do you plan to do a new release

**Pull request merged**